pymc3/step_methods/sgmcmc.py

```diff
@@ -42,12 +42,13 @@ def elemwise_dlogL(vars, model, flat_view):
     # tensor of shape (batch_size,)
     logL = obs_var.logp_elemwiset.sum(axis=tuple(range(1, obs_var.logp_elemwiset.ndim)))
     # calculate fisher information
-    terms = tt.concatenate([\
-                theano.scan(lambda i, logX: theano.grad(logX[i], v).flatten(),\
+    terms = []
+    for var in vars:
```

Comment:
this doesn't fix it -- the lambda will still look up `var` when it is first used.  you'll need to change the signature of the lambda to something like `lambda i, logX, v=var: ...`  This will immediately bind the value of `var` to `v` as the default variable, because that's how python rolls.